### PR TITLE
python: replace suricatasc and suricatactl with rust - v1

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/8571
Ticket: https://redmine.openinfosecfoundation.org/issues/6287

Just tossing this out there for discussion again. I feel that I'm drifting further away from Python, and I also feel that Python more and more requires deployment in controlled environments like virtualenv or containers making it less suitable than it was for general-purpose apps running in a variety of environments.

### CODE REVIEW

File: rust/client/examples/iface-list.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args: Vec<String> = std::env::args().collect();
+
+    let filename = if let Some(filename) = args.get(1) {
+        filename
+    } else {
+        "/run/suricata/suricata-command.socket"
+    };
+    dbg!(filename);
```
Comment (jasonish) [inline rust/client/examples/iface-list.rs:17]: Oops, debug.

File: rust/client/examples/iface-list.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    let mut client = Client::connect(filename, false)?;
+    client.send(&json!({"command": "iface-list"}))?;
+    let response = client.read()?;
+    dbg!(response);
```
Comment (jasonish) [inline rust/client/examples/iface-list.rs:22]: And again.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16577

**Comment by catenacyber** [comment]
I like the written in rust.
What is there to discuss ?

**Comment by jasonish** [comment]
Will rebase.

**Pull request closed without merging**